src/internal/kopia/upload.go

```diff
@@ -991,6 +990,11 @@ func inflateBaseTree(
 			return clues.Wrap(err, "subtree root is not directory").WithClues(ictx)
 		}
 
+		newSubtreePath := subtreePath
```

Comment:
nit: add a comment that this handles migration of the prefix directories. That way the comment below that explains that it assumes no prefix directory changes makes sense (idk how I missed that earlier 🙃)